packages/api/lambdas/cleanExecutions.js

```diff
@@ -1,48 +1,248 @@
+      else if (cleanupOnlyNonRunning) queryBuilder.where('status', '!=', 'running');
+    })
+    .limit(updateLimit);
+
+  // this is done as a search:update because postgres doesn't support limited updates
+  await knex(executionModel.tableName)
+    .whereIn('cumulus_id', executionIds.map((execution) => execution.cumulus_id))
+    .update(wipedPayloads);
+};
+
+/**
+ * parse out environment variable configuration
+ * @returns {{
+ *   cleanupNonRunning: boolean,
+ *   cleanupRunning: boolean,
+ *   cleanupPostgres: boolean,
+ *   cleanupES: boolean,
+ *   payloadTimeout: number
+ *   esIndex: string,
+ *   updateLimit: number,
+ * }}
+ */
+const parseEnvironment = () => {
+  const cleanupNonRunning = JSON.parse(process.env.CLEANUP_NON_RUNNING || 'true');
+  const cleanupRunning = JSON.parse(process.env.CLEANUP_RUNNING || 'false');
+
+  const cleanupPostgres = JSON.parse(process.env.CLEANUP_POSTGRES || 'true');
+  const cleanupES = JSON.parse(process.env.CLEANUP_ES || 'true');
+  if (!cleanupRunning && !cleanupNonRunning) throw new Error('running and non-running executions configured to be skipped, nothing to do');
+  if (!cleanupES && !cleanupPostgres) throw new Error('elasticsearch and postgres executions configured to be skipped, nothing to do');
+
+  const _payloadTimeout = process.env.PAYLOAD_TIMEOUT || '10';
+  const payloadTimeout = Number.parseInt(_payloadTimeout, 10);
+  if (!Number.isInteger(payloadTimeout)) {
+    throw new TypeError(`Invalid number of days specified in configuration for payloadTimeout: ${_payloadTimeout}`);
+  }
+  const esIndex = process.env.ES_INDEX || 'cumulus';
+
+  const updateLimit = Number(process.env.UPDATE_LIMIT || 10000);
+  return {
+    cleanupRunning,
+    cleanupNonRunning,
+    cleanupPostgres,
+    cleanupES,
+    payloadTimeout,
+    esIndex,
+    updateLimit,
+  };
+};
 
-  const execution = new ExecutionModel();
-  return await execution.removeOldPayloadRecords(completeTimeout,
-    nonCompleteTimeout,
-    completeDisable,
-    nonCompleteDisable);
+/**
+ * parse environment variables to extract configuration and run cleanup of PG and ES executions
+ *
+ * @returns {Promise<void>}
+ */
+async function cleanExecutionPayloads() {
+  const envConfig = parseEnvironment();
+  log.info(`running cleanExecutions with configuration ${JSON.stringify(envConfig)}`);
+  const {
+    updateLimit,
+    cleanupRunning,
+    cleanupNonRunning,
+    cleanupPostgres,
+    cleanupES,
+    payloadTimeout,
+    esIndex,
+  } = envConfig;
+
+  const promises = [];
+  if (cleanupES) {
+    promises.push(cleanupExpiredESExecutionPayloads(
+      payloadTimeout,
+      cleanupRunning,
+      cleanupNonRunning,
+      updateLimit,
+      esIndex
+    ));
+  }
+  if (cleanupPostgres) {
+    promises.push(cleanupExpiredPGExecutionPayloads(
+      payloadTimeout,
+      cleanupRunning,
+      cleanupNonRunning,
+      updateLimit
+    ));
+  }
+  await Promise.all(promises);
```

Comment:
Just want to make sure this is where we landed. If either promise rejects, we'll stop, right? So if ES cleanup is first in the list and fails, we won't attempt PG. Is that what we decided or do we want them idempotent? 

Allowing one and not the other is 👍

Reply:
I feel like you're misreading this? if not then *I'm* misunderstanding this and please clarify what's happening here.
cleanupExpiredPg launches without awaiting, cleanupExpiredES Launches without awaiting, and then we await both promises right?

Reply:
made irrelevant by removing PG